FirebaseVertexAI/Sources/VertexAI.swift

```diff
@@ -0,0 +1,112 @@
+// limitations under the License.
+
+import Foundation
+
+import FirebaseAppCheckInterop
+import FirebaseCore
+
+// Exports the GoogleGenerativeAI module to users of the SDK.
+@_exported import GoogleGenerativeAI
+
+// Avoids exposing internal FirebaseCore APIs to Swift users.
+@_implementationOnly import FirebaseCoreExtension
+
+@available(iOS 15.0, macOS 11.0, macCatalyst 15.0, *)
+@objc(FIRVertexAI)
+open class VertexAI: NSObject {
+  // MARK: - Public APIs
+
+  /// Returns an instance of `GoogleGenerativeAI.GenerativeModel` that uses the Vertex AI API.
+  ///
+  /// This instance is configured with the default `FirebaseApp`.
+  public static func generativeModel(modelName: String, location: String) -> GoogleGenerativeAI
+    .GenerativeModel {
+    return generativeModel(app: FirebaseApp.app()!, modelName: modelName, location: location)
+  }
+
+  /// Returns an instance of `GoogleGenerativeAI.GenerativeModel` that uses the Vertex AI API.
+  public static func generativeModel(app: FirebaseApp, modelName: String,
+                                     location: String) -> GoogleGenerativeAI.GenerativeModel {
+    guard let provider = ComponentType<VertexAIProvider>.instance(for: VertexAIProvider.self,
+                                                                  in: app.container) else {
+      fatalError("No \(VertexAIProvider.self) instance found for Firebase app: \(app.name)")
+    }
+    let modelResourceName = modelResourceName(app: app, modelName: modelName, location: location)
+    let vertexAI = provider.vertexAI(location: location, modelResourceName: modelResourceName)
+
+    return vertexAI.model
+  }
+
+  // MARK: - Private
+
+  /// The `FirebaseApp` associated with this `VertexAI` instance.
+  private let app: FirebaseApp
+
+  private let appCheck: AppCheckInterop?
+
+  private let location: String
+
+  private let modelResouceName: String
+
+  lazy var model: GenerativeModel = {
+    let options = RequestOptions(
+      apiVersion: "v2beta",
+      endpoint: "staging-firebaseml.sandbox.googleapis.com",
+      hooks: [addAppCheckHeader]
+    )
+    return GenerativeModel(
+      name: modelResouceName,
+      apiKey: app.options.apiKey!,
+      requestOptions: options
+    )
+  }()
+
+  init(app: FirebaseApp, location: String, modelResourceName: String) {
+    self.app = app
+    appCheck = ComponentType<AppCheckInterop>.instance(for: AppCheckInterop.self, in: app.container)
+    self.location = location
+    modelResouceName = modelResourceName
+  }
+
+  private static func modelResourceName(app: FirebaseApp, modelName: String,
+                                        location: String) -> String {
+    if modelName.contains("/") {
+      return modelName
+    }
+    guard let projectID = app.options.projectID else {
+      print("The FirebaseApp is missing a project ID.")
+      return modelName
+    }
+
+    return "projects/\(projectID)/locations/\(location)/publishers/google/models/\(modelName)"
```

Comment:
Since `location` isn't an optional parameter I added a check for `""`. Beyond that I'll leave it up to the backend to return a 404 error if an invalid location is specified.